### ISSUE TITLE
Where is texture format specified?

### ISSUE THREAD

**Issue by dakom** [KhronosGroup/glTF#1280]
From looking at previous issues such as [this one](https://github.com/KhronosGroup/glTF/issues/835), it seems that there is a property `texture.format`. However, the spec doesn't list that as far as I see (neither [here](https://github.com/KhronosGroup/glTF/tree/master/specification/2.0#reference-texture) nor [here](https://github.com/KhronosGroup/glTF/tree/master/specification/2.0#texture-data))

Additionally, it seems as though the format is determined by the specific material. For example:

> **material.emissiveTexture**
> This texture contains RGB components in sRGB color space. 

> **material.normalTexture**
> The texture contains RGB components in linear space.

> **pbrMetallicRoughness.baseColorTexture**
> This texture contains RGB(A) components in sRGB color space

Based on this- I'm a bit confused and would love some help understanding this better:

1. is "linear space" equivilent to gl.RGB/gl.RGBA? (e.g. for loading `normalTexture`)

2. The language for `pbrMetallicRoughness.metallicRoughnessTexture` could be a little clearer... it does say "these values are linear", but it might be helpful to use the same consistent language of "This texture contains components in x color space"

3. It feels natural to load a texture with all of the appropriate settings, and then reference it from a material, but the above setup requires crawling through the tree to find the material which references the texture, in order to determine the settings... I guess this isn't such a problem, but it does feel a little bit weird - wouldn't it feel a bit better to simply specify that each texture has a `format` field? (and then a sanity check should confirm that the texture pointed to by each material has the correct color space)

Long story short - all the above would be resolved by a `texture.format` field, but along with not seeing it in the spec, I'm also not seeing it in the demo gltf files... am I missing something? :)

**Comment by ale64bit** [comment]
There are specific format descriptions for [normalTexture](https://github.com/KhronosGroup/glTF/blob/master/specification/2.0/README.md#reference-normaltextureinfo) and [occlusionTexture](https://github.com/KhronosGroup/glTF/blob/master/specification/2.0/README.md#reference-occlusiontextureinfo). There's also a generic [textureInfo](https://github.com/KhronosGroup/glTF/blob/master/specification/2.0/README.md#reference-textureinfo) format description. However, I can't find the format spec for `emissiveTexture` anywhere.

I do agree that the spec could be more clear about this.

**Comment by dakom** [comment]
@ale64bit - right on, but isn't it a bit awkward to determine whether a texture is a `normalTexture` or `occlusionTexture` ? Unless I'm mistaken - you need to do something like a reverse lookup, walking the list of materials until you find something like `material.normalTexture.index === textureIndex`

**Comment by aguaviva** [comment]
The images section should specify whether the texture should be interpreted as:
- RGB (metallic, norma maps)
- SRGB (emmisive, albedo)

That way the textures can be loaded by parsing a minimal amount of the json file, otherwise we need to do a reverse lookup as dakom @dakom.
